src/random.cpp

```diff
@@ -129,20 +130,87 @@ void GetRandBytes(unsigned char* buf, int num)
     }
 }
 
+
+/* /fn int getRDSEED( uint64_t *buff, int num)
+ * /brief Use Intel's SP 800-90B & C compliant Hardware implemented
+ *        instruction to add entropy to the pool use to create keys. 
+ *
+ *        For details, see: https://software.intel.com/content/www/us/en/develop/blogs/the-difference-between-rdrand-and-rdseed.html
+ * 
+ *  /param buff Buffer to fill with random bits.
+ *  /param num  Number of 64-bit words to write to buffer.
+ *  /return     Returns 1 if the hardware returned  a valid stream of random numbers, 0 otherwise.
+ *
+ */
+int getRDSEED( uint64_t *buff, uint32_t num){
+    //Initialize state to 0, or failed.
+    uint32_t  cumulativeStatus = 0;
+
+    #ifdef LOWELEVEL_USE_RDSEEED
+        //Check to see if the processor running this code has 
+        //this Instruction. Most Intel/AMD CPUs after 2015
+        //should have this instruction.
+        uint32_t level, eax, ebx, ecx, edx; 
+        
+        //Level where the RDSEED feature is described
+        level = 1;
+    
+        //Extract feature vector
+        __get_cpuid(level, &eax, &ebx, &ecx, &edx);    
+    
+        //Check if RDSEED exists on this processor
+        if( (ecx & bit_RDSEED) == bit_RDSEED ) {
+    
+            //Re-initialize value to 1, or success 
+            cumulativeStatus = 1;
+    
+            for( uint32_t kk = 0; kk < num; kk++){
+                //Get RBG data
+                unsigned int rng_high, rng_low;
+
+                int32_t status1 = _rdseed32_step( &rng_low );
+                int32_t status2 = _rdseed32_step( &rng_high );
```

Comment:
Should place in a retry loop
https://software.intel.com/content/www/us/en/develop/articles/intel-digital-random-number-generator-drng-software-implementation-guide.html#inpage-nav-5-9

Reply:
The entire function returns a 1 or 0 on success or failure, respectively, it is the caller's responsibility to address that properly. In our case, we have. Look at the calling code, if it fails, we move on without using it.

Reply:
The processor can request bits faster than the entropy source can output bits, so it would be ideal to retry until new bits are available instead of discarding all the bits due to one or more failures.

Reply:
The question becomes how often should we try before we give up, how many cycles should we wait to retry again, in short how much time is it okay to wait until we get the requested random bits? As it is, if it is not ready when we ask, we move on. I think spending more than 1,000 clock cycles on this is too much. We could run the PAUSE  and WAIT instruction and then try again, though the behavior of the PAUSE instruction is very processor specific in terms of how many cycles it delays instruction execution, but as it is I think it is good enough for now. Plus, cpuid is called before this, so between calls of our function  there is a very long delay in cycles while cpuid clears the execution pipeline, stalls the reorder buffer and waits for memory operations instructions to be retired. We only ask for 512-bits of entropy, and this function does not get called with high frequency. Who generates thousands of wallets a second anyway?